### DISCUSSION TITLE
Remaining time parameter

### DISCUSSION THREAD

**Discussion by Lecso11** [supermerill/SuperSlicer#2643] [Q&A]
I'm trying to make a custom gcode for before layer change gcode, that writes on the screen the current layer, total layer number, and remaining time.
I'm using klipper, I'm passing my variables to my macro, that does the actual writing.

My before layer change code at the moment is:
`DISPLAY_LAYER_INFO LAYER={layer_num + 1} TOTAL={total_layer_count} R={}`

My macro is:
```
[gcode_macro DISPLAY_LAYER_INFO]
gcode:
    {% set current_layer = params.LAYER|int %}
    {% set total_layer = params.TOTAL|int %}
    {% set remaining_time = params.R|int %}

    M117 {current_layer} of {total_layer},  { "%02d:%02d" % (remaining_time // 60, remaining_time % 60) } remaining
```
My problem is, I don't know what variable to pass as R.

**Comment by neophyl** [comment]
Might want to read https://github.com/supermerill/SuperSlicer/issues/2121
